// @ionic-native/taptic-engine/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var TapticEngine = /** @class */ (function (_super) {
    __extends(TapticEngine, _super);
    function TapticEngine() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TapticEngine.prototype.selection = function () { return cordova(this, "selection", {}, arguments); };
    TapticEngine.prototype.notification = function (options) { return cordova(this, "notification", {}, arguments); };
    TapticEngine.prototype.impact = function (options) { return cordova(this, "impact", {}, arguments); };
    TapticEngine.prototype.gestureSelectionStart = function () { return cordova(this, "gestureSelectionStart", {}, arguments); };
    TapticEngine.prototype.gestureSelectionChanged = function () { return cordova(this, "gestureSelectionChanged", {}, arguments); };
    TapticEngine.prototype.gestureSelectionEnd = function () { return cordova(this, "gestureSelectionEnd", {}, arguments); };
    TapticEngine.pluginName = "TapticEngine";
    TapticEngine.plugin = "cordova-plugin-taptic-engine";
    TapticEngine.pluginRef = "TapticEngine";
    TapticEngine.repo = "https://github.com/EddyVerbruggen/cordova-plugin-taptic-engine";
    TapticEngine.platforms = ["iOS"];
TapticEngine.ɵfac = function TapticEngine_Factory(t) { return ɵTapticEngine_BaseFactory(t || TapticEngine); };
TapticEngine.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: TapticEngine, factory: function (t) { return TapticEngine.ɵfac(t); } });
var ɵTapticEngine_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(TapticEngine);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TapticEngine, [{
        type: Injectable
    }], null, null); })();
    return TapticEngine;
}(IonicNativePlugin));
export { TapticEngine };
