// ng2-truncate/__ivy_ngcc__/dist/truncate-characters.pipe.js
import { Pipe } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var TruncateCharactersPipe = (function () {
    function TruncateCharactersPipe() {
    }
    TruncateCharactersPipe.prototype.transform = function (value, limit, trail) {
        if (limit === void 0) { limit = 40; }
        if (trail === void 0) { trail = '…'; }
        if (!value) {
            value = '';
        }
        if (limit < 0) {
            limit *= -1;
            return value.length > limit ? trail + value.substring(value.length - limit, value.length) : value;
        }
        else {
            return value.length > limit ? value.substring(0, limit) + trail : value;
        }
    };
    /** @nocollapse */
    TruncateCharactersPipe.ctorParameters = function () { return []; };
TruncateCharactersPipe.ɵfac = function TruncateCharactersPipe_Factory(t) { return new (t || TruncateCharactersPipe)(); };
TruncateCharactersPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "truncate", type: TruncateCharactersPipe, pure: true });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TruncateCharactersPipe, [{
        type: Pipe,
        args: [{
                name: 'truncate'
            }]
    }], function () { return []; }, null); })();
    return TruncateCharactersPipe;
}());
export { TruncateCharactersPipe };
