// ng2-truncate/__ivy_ngcc__/dist/truncate.module.js
import { NgModule } from '@angular/core';
import { TruncateCharactersPipe } from './truncate-characters.pipe';
import { TruncateWordsPipe } from './truncate-words.pipe';
import * as ɵngcc0 from '@angular/core';
export var TRUNCATE_PIPES = [TruncateCharactersPipe, TruncateWordsPipe];
var TruncateModule = (function () {
    function TruncateModule() {
    }
    /** @nocollapse */
    TruncateModule.ctorParameters = function () { return []; };
TruncateModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TruncateModule });
TruncateModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TruncateModule_Factory(t) { return new (t || TruncateModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TruncateModule, { declarations: [TruncateCharactersPipe, TruncateWordsPipe], exports: [TruncateCharactersPipe, TruncateWordsPipe] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TruncateModule, [{
        type: NgModule,
        args: [{
                declarations: [TRUNCATE_PIPES],
                exports: [TRUNCATE_PIPES]
            }]
    }], function () { return []; }, null); })();
    return TruncateModule;
}());
export { TruncateModule };
